### PR TITLE
actually warn, don't raise on duplicate ids in LiveViewTest

### PR DESCRIPTION
**Pull request by SteffenDE**
Relates to #3588.
Relates to: https://github.com/phoenixframework/phoenix_live_view/pull/3560.

cc @josevalim

### CODE REVIEW

File: lib/phoenix_live_view/test/dom.ex [hunk 1]
```diff
@@ -44,17 +44,19 @@ defmodule Phoenix.LiveViewTest.DOM do
...
+
+          #{inspect_html(node)}
 
           LiveView requires that all elements have unique ids, duplicate IDs will cause
           undefined behavior at runtime, as DOM patching will not be able to target the correct
           elements.
-          """
-        else
-          detect_duplicate_ids(children, MapSet.put(ids, id))
+          """)
```
Comment (josevalim) [inline lib/phoenix_live_view/test/dom.ex:55]: We probably want to pass `[]` as second argument? We don't have a useful stacktrace here, do we?

Comment (SteffenDE) [reply lib/phoenix_live_view/test/dom.ex]: sadly no good stacktrace. Changed!

Comment (SteffenDE) [reply lib/phoenix_live_view/test/dom.ex]: actually, at least it contains the test, so maybe we should keep it?

```
warning: Duplicate id found while testing LiveView: foo

    <div id="foo"></div>


LiveView requires that all elements have unique ids, duplicate IDs will cause
undefined behavior at runtime, as DOM patching will not be able to target the correct
elements.

  (phoenix_live_view 1.0.1) lib/phoenix_live_view/test/dom.ex:47: Phoenix.LiveViewTest.DOM.detect_duplicate_ids/2
  (phoenix_live_view 1.0.1) lib/phoenix_live_view/test/dom.ex:36: Phoenix.LiveViewTest.DOM.detect_duplicate_ids/2
  (phoenix_live_view 1.0.1) lib/phoenix_live_view/test/dom.ex:28: Phoenix.LiveViewTest.DOM.parse/1
  test/phoenix_live_view/test/dom_warn_test.exs:10: Phoenix.LiveViewTest.DOMWarnTest."test parse detects duplicate ids"/1
  (ex_unit 1.19.0-dev) lib/ex_unit/runner.ex:511: ExUnit.Runner.exec_test/2
  (stdlib 6.1.2) timer.erl:590: :timer.tc/2
  (ex_unit 1.19.0-dev) lib/ex_unit/runner.ex:433: anonymous fn/6 in ExUnit.Runner.spawn_test_monitor/4
```

Comment (josevalim) [reply lib/phoenix_live_view/test/dom.ex]: Yeah, we keep it!

**Review by josevalim — APPROVED** [review]

**Pull request merged**